### DISCUSSION TITLE
Can I use Visitors for light personalization?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#2510] [Q&A] [answered]
I have a project that requires remembering some basic visitor information (like first time they have visited) without having logins.  I could a drop a cookie and track that in a table, but then I saw the visitor table and thought it might be doing that for me.  Ha anyone used the visitor table in this way?

**Comment by sbwalker** [comment]
This is exactly what the Visitor service is intended for. It can be used in conjunction with the Settings service to store custom personalization attributes related to a visitor.

**Comment by markdav-is** [comment]
This thread might be dated, but the level of detail we’re expecting in our visitor data isn’t quite there. I’ll investigate how our code determines a unique visitor, as we’ve noticed identical data appearing on separate computers within the same office. We’ve developed a shopping cart module that doesn’t necessitate a login, and we’re using PageState.VisitorId to distinguish unique visitors. During testing, our QA team has reported that they can perform the following actions:

computer 1: add items to cart.
computer 2: see items that were added by computer 1

The only fields recorded in the visitor table are IPAddress and UserAgent.  this leads me to believe that all PCs behind the same firewall would be seen as the same visitor.

**Reply by sbwalker** [reply, marked as answer]
@markdav-is if you go to Visitor Management and click on the Settings tab you will see a Correlation option which is enabled by default. This option controls whether users from the same IP should be treated as the same visitor or different visitors. 

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/edcdec55-3f51-44f9-b7ea-9870d8a31c2d)

**Reply by markdav-is** [reply]
Thank you so much!  I love being able to fix this issue with a setting!
